src/commands/ssh-resolve.ts

```diff
@@ -0,0 +1,119 @@
+import fs from "fs";
+import path from "path";
+import tmp from "tmp-promise";
+import yargs from "yargs";
+
+export const sshResolveCommand = (yargs: yargs.Argv) =>
+  yargs.command<SshResolveCommandArgs>(
+    "ssh-resolve <destination>",
+    "SSH into a virtual machine",
+    (yargs) =>
+      yargs
+        .positional("destination", {
+          type: "string",
+          demandOption: true,
+        })
+        .option("parent", {
+          type: "string",
+          describe:
+            "The containing parent resource which the instance belongs to (account, project, subscription, etc.)",
+        })
+        .option("provider", {
+          type: "string",
+          describe: "The cloud provider where the instance is hosted",
+          choices: ["aws", "azure", "gcloud"],
+        })
+        .option("debug", {
+          type: "boolean",
+          describe: "Print debug information.",
+        })
+        .option("quiet", {
+          alias: "q",
+          type: "boolean",
+          describe: "Suppress output",
+        }),
+
+    fsShutdownGuard(sshResolveAction)
+  );
+
+/** Connect to an SSH backend
+ *
+ * Implicitly gains access to the SSH resource if required.
+ *
+ * Supported SSH mechanisms:
+ * - AWS EC2 via SSM with Okta SAML
+ */
+const sshResolveAction = async (
+  args: yargs.ArgumentsCamelCase<SshResolveCommandArgs>
+) => {
+  const p0Executable = bootstrapConfig.appPath;
+  // Prefix is required because the backend uses it to determine that this is an AWS request
+  const authn = await authenticate({ noRefresh: args.quiet ?? false }).catch(
+    conditionalAbortBeforeThrow(args.quiet ?? false)
+  );
+
+  const { request, provisionedRequest } = await prepareRequest(
+    authn,
+    args,
+    args.destination,
+    true,
+    args.quiet
+  ).catch(conditionalAbortBeforeThrow(args.quiet ?? false));
+
+  const sshProvider = SSH_PROVIDERS[provisionedRequest.permission.provider];
+
+  const keys = await sshProvider?.generateKeys?.(
+    provisionedRequest.permission.resource,
+    {
+      debug: args.debug,
+    }
+  );
+
+  const tmpFile = tmp.fileSync();
+  fs.writeFileSync(tmpFile.name, JSON.stringify(request, null, 2));
+
+  const identityFile =
+    keys?.privateKeyPath ?? path.join(P0_PATH, "ssh", "id_rsa");
+  const certificateInfo = keys?.certificatePath
+    ? `CertificateFile ${keys.certificatePath}`
+    : "";
+
+  const data = `
+Hostname ${args.destination}
+  User ${request.linuxUserName}
+  IdentityFile ${identityFile}
+  ${certificateInfo}
+  PasswordAuthentication no
+  ProxyCommand ${p0Executable} ssh-proxy %h --port %p --provider ${provisionedRequest.permission.provider} --identity-file ${identityFile} --request-json ${tmpFile.name}`;
+
+  await fs.promises.mkdir(path.join(P0_PATH, "ssh", "configs"), {
+    recursive: true,
+  });
+  const configLocation = path.join(
+    P0_PATH,
+    "ssh",
+    "configs",
+    `${args.destination}.config`
+  );
+
+  fs.writeFileSync(configLocation, data);
```

Comment:
we should add some debug logs for this command so that it gets easier to troubleshoot, i was having a hard time connecting and removing the `-q` flag didn't provide enough clarity about why my configs weren't matching.